cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go

```diff
@@ -673,3 +675,53 @@ func newFluxHelmRelease(chart *models.Chart, targetName types.NamespacedName, ve
 	}
 	return &unstructuredRel, nil
 }
+
+// resourceRefsFromManifest returns the resource refs for a given yaml manifest.
+func resourceRefsFromManifest(m string) ([]*corev1.ResourceRef, error) {
+	decoder := yaml.NewYAMLToJSONDecoder(strings.NewReader(m))
+	refs := []*corev1.ResourceRef{}
+	doc := yamlResource{}
+	for {
+		err := decoder.Decode(&doc)
+		if err != nil {
+			if goerrs.Is(err, io.EOF) {
+				break
+			}
+			return nil, status.Errorf(codes.Internal, "Unable to decode yaml manifest: %v", err)
+		}
+		if doc.Kind == "" {
+			continue
+		}
+		if doc.Kind == "List" || doc.Kind == "RoleList" || doc.Kind == "ClusterRoleList" {
+			for _, i := range doc.Items {
+				refs = append(refs, &corev1.ResourceRef{
+					ApiVersion: i.APIVersion,
+					Kind:       i.Kind,
+					Name:       i.Metadata.Name,
+					Namespace:  i.Metadata.Namespace,
+				})
+			}
+			continue
+		}
+		refs = append(refs, &corev1.ResourceRef{
+			ApiVersion: doc.APIVersion,
+			Kind:       doc.Kind,
+			Name:       doc.Metadata.Name,
+			Namespace:  doc.Metadata.Namespace,
+		})
+	}
+
+	return refs, nil
+}
+
+type yamlMetadata struct {
```

Comment:
that and the next type feel like they belong in some pkg to be re-used by multiple components? There's nothing flux-specific about this

Reply:
Yep, I mentioned in the description above that we may want to share functionality between the helm and flux plugins. Could be even broader than that.